packages/gatsby-source-drupal/src/normalize.js

```diff
@@ -6,7 +6,7 @@ const nodeFromData = (datum, createNodeId) => {
   const preservedId =
     typeof _attributes_id !== `undefined` ? { _attributes_id } : {}
   return {
-    id: createNodeId(datum.id),
+    id: createNodeId(`${datum.attributes.langcode}${datum.id}`),
```

Comment:
At first glance, it seems like all entities have a `attributes.langcode` attribute. However, I have found that for some reason JSONAPI does not always return this langcode value if you are browsing as an anonymous user. For an example try going to `/jsonapi/date_format/date_format` as an authenticated and an anonymous user. For me I see the langcode when logged in but not when anonymous. This breaks builds if the user is not using authentication.

The simple workaround here is to just check if the langcode exists and if it doesn't, default to `und`

Reply:
I think it is related to fact that date_format is config entity in Drupal. So it may be related to enabling config_translation module.

Reply:
There's quite a few entities that seem to have no langcode, mostly internal entities. I'll make sure to review this during my next pass.

Reply:
Basically there are 2 types of entities in Drupal - content(node, user...) and config(node type, language...).
We have 2 modules: content_translation and config_translation. So langcode property means language of that entity.
For example we can have language entity.
id = cs, langcode = en, label = Czech
id = cs, langcode = cs, label = Česky
id = en, langcode = en, label = English
id = en, langcode = cs, label = Anglicky

From my point of view: i expose only content entities. I do not see reason to expose config in general. Being developer i prefer langConfig in gatsby-config.js. Why to store this in Gatsby index?

If i wanna build page with News i expose only node--article resource in jsonapi(i use jsonapi_extras module in Drupal). I use `included` (either in Gatsby or in Drupal with jsonapi_defaults) for adding related data like author and images into News.
(i prefer this way compared to `disallowedLinkTypes` of Gatsby.

Reply:
I guess this is a valid point, especially for larger sites you'll be pulling a lot of unnecessary entities that might not be shown on your front end. I think that'd be an idea for another ticket. Perhaps an "allowedLinkTypes" option which, when set, whitelists particular entities types?

Reply:
TODO:
- Check if langcode exists and if it doesn't use `und`

Reply:
Still relevant.